plugins/com.google.cloud.tools.eclipse.test.util/src/com/google/cloud/tools/eclipse/test/util/BasePluginXmlTest.java

```diff
@@ -189,6 +192,35 @@ public final void testBundleVendor() throws IOException {
     assertEquals("Google Inc.", vendor);
   }
 
+  @Test
+  public final void testGuavaImportVersions() throws IOException {
+    checkVersionStringsInManifestMf(
```

Comment:
This looks really useful.  Could I suggest a refactoring?
  - call this `testDependencyVersions()`
  - add methods `checkImportPackageVersion(package-pattern, version-range)` and `checkRequireBundleVersion(bundle-name, version-range)` (e.g., `checkImportPackageVersion("com.google.common.*", "[20,21)")`) — embedding `"Require-Bundle"` or `"Import-Package"` and `"version"` vs `"bundle-version"` is error prone
  - use `org.osgi.framework.VersionRange#valueOf(String)` to parse and compare the version ranges
  - the patterns should allow `\s` for white-space

Reply:
Good idea. But I am not thrilled to spend more time on this to make it general and flexible ATM. If there is unnecessary white-space, let's just remove it and be consistent.